From 13bd3cd67e3c0efdf8bb578332ccf615f51208aa Mon Sep 17 00:00:00 2001
From: Yotam Ben Dosa <ybendosa@habana.ai>
Date: Tue, 05 Apr 2022 13:07:44 +0300
Subject: [PATCH] [SW-0] hls: make openbmc cmd wrappers common

Since both HLS-1 and HLS-3 are able to use these wrappers, it's best to
make them common, and not duplicate the implementation.

Signed-off-by: Yotam Ben Dosa <ybendosa@habana.ai>
Change-Id: I8c981aeed7ba962c5bac4acf2b45da774327a97e
---

diff --git a/scripts/make/common/Makefile.openbmc b/scripts/make/common/Makefile.openbmc
new file mode 100644
index 0000000..08e5711
--- /dev/null
+++ b/scripts/make/common/Makefile.openbmc
@@ -0,0 +1,119 @@
+U_HL_DEV=$(shell echo ${HL_DEV} | tr '[:lower:]' '[:upper:]')
+
+openbmc-help:
+	@echo
+	@echo 'OpenBMC bitbake & devtool wrappers for Habana Labs server build'
+	@echo '---------------------------------------------------------------'
+	@echo
+	@echo 'OpenBMC devtool commands can be used for adding new'
+	@echo ' packages and modifying existing ones, these targets wrap its'
+	@echo ' functionality:'
+	@echo
+	@echo '1. openbmc-help:'
+	@echo '   Print this help menu.'
+	@echo '2. openbmc-add:'
+	@echo '   Adds a new package from source, and generates a recipe for it.'
+	@echo '   Usage*:'
+	@echo '   $ make openbmc-add URI="<git://[...]>"'
+	@echo '   * In case of ssh repos, ";protocol=ssh" should be added as a'
+	@echo '     suffix.'
+	@echo '3. openbmc-<pkg-name>-modify:'
+	@echo '   Deployes a package already managed by OpenBMC and opens a'
+	@echo '   terminal in its sources path.'
+	@echo '   From now on, each change in the sources will trigger'
+	@echo '   a new build when calling 'make $(HL_DEV)-build'.'
+	@echo '   Usage:'
+	@echo '   $ make openbmc-<pkg-name>-modify'
+	@echo '4. openbmc-<pkg-name>-menuconfig:'
+	@echo '   Opens a 'menuconfig' session for the package'
+	@echo '   (if it has a menuconfig target).'
+	@echo '   Usage:'
+	@echo '   $ make openbmc-<pkg-name>-menuconfig'
+	@echo '5. openbmc-<pkg-name>-build:'
+	@echo '   Builds a package that is already deployed via'
+	@echo '   openbmc-<pkg-name>-modify.'
+	@echo '   Usage:'
+	@echo '   $ make openbmc-<pkg-name>-build'
+	@echo '6. openbmc-<pkg-name>-clean:'
+	@echo '   Cleans the build artifacts of the package.'
+	@echo '   Usage:'
+	@echo '   $ make openbmc-<pkg-name>-clean'
+	@echo '7. openbmc-<pkg-name>-remove*:'
+	@echo '   Clears the package files from the local environment*.'
+	@echo '   Usage:'
+	@echo '   $ make openbmc-<pkg-name>-remove'
+	@echo '   * The sources dir will be deleted, so one needs to make'
+	@echo '     sure his work was pushed before running it.'
+	@echo '8. openbmc-<pkg-name>-diffconfig:'
+	@echo '   Opens a 'diffconfig' session for the package'
+	@echo '   (if it has a diffconfig target).'
+	@echo '   Usage:'
+	@echo '   $ make openbmc-<pkg-name>-diffconfig'
+
+inner-openbmc-add:
+	@cd $($(U_HL_DEV)_DEPLOY_DIR);			\
+	$(U_HL_DEV)_ENV $($(U_HL_DEV)_BITBAKE_ENV)	\
+	devtool add "$(URI)"
+
+openbmc-add: $(HL_DEV)-asic-prepare
+	@$(MAKE) inner-$@
+
+inner-openbmc-%-modify:
+	@cd $($(U_HL_DEV)_DEPLOY_DIR);					\
+	$($(U_HL_DEV)_ENV) $($(U_HL_DEV)_BITBAKE_ENV)			\
+	devtool modify $(patsubst inner-openbmc-%-modify,%,$@);		\
+	terminator							\
+	--working-directory="$($(U_HL_DEV)_DEVTOOL_SOURCES)/$(patsubst	\
+		inner-openbmc-%-modify,%,$@)" &
+
+openbmc-%-modify: $(HL_DEV)-asic-prepare
+	@$(MAKE) inner-$@
+
+inner-openbmc-%-menuconfig:
+	@cd $($(U_HL_DEV)_DEPLOY_DIR);					\
+	$($(U_HL_DEV)_ENV $($(U_HL_DEV)_BITBAKE_ENV))			\
+	bitbake -c menuconfig $(patsubst inner-openbmc-%-menuconfig,%,$@)
+
+openbmc-%-menuconfig: $(HL_DEV)-asic-prepare
+	@$(MAKE) inner-$@
+
+inner-openbmc-%-diffconfig:
+	@cd $($(U_HL_DEV)_DEPLOY_DIR);					\
+	$($(U_HL_DEV)_ENV) $($(U_HL_DEV)_BITBAKE_ENV)			\
+	bitbake -c diffconfig $(patsubst inner-openbmc-%-diffconfig,%,$@)
+
+openbmc-%-diffconfig: $(HL_DEV)-asic-prepare
+	@$(MAKE) inner-$@
+
+inner-openbmc-%-build:
+	@cd $($(U_HL_DEV)_DEPLOY_DIR);					\
+	$($(U_HL_DEV)_ENV) $($(U_HL_DEV)_BITBAKE_ENV)			\
+	devtool build $(patsubst inner-openbmc-%-build,%,$@)
+
+openbmc-%-build: $(HL_DEV)-asic-prepare
+	@$(MAKE) inner-$@
+
+inner-openbmc-%-clean:
+	@cd $($(U_HL_DEV)_DEPLOY_DIR);					\
+	$($(U_HL_DEV)_ENV) $($(U_HL_DEV)_BITBAKE_ENV)			\
+	bitbake -c clean $(patsubst inner-openbmc-%-clean,%,$@)
+
+openbmc-%-clean: $(HL_DEV)-asic-prepare
+	@$(MAKE) inner-$@
+
+inner-openbmc-%-cleanall:
+	@cd $($(U_HL_DEV)_DEPLOY_DIR);					\
+	$($(U_HL_DEV)_ENV) $($(U_HL_DEV)_BITBAKE_ENV)			\
+	bitbake -c cleanall $(patsubst inner-openbmc-%-cleanall,%,$@)
+
+openbmc-%-cleanall: $(HL_DEV)-asic-prepare
+	@$(MAKE) inner-$@
+
+openbmc-%-remove:
+	@cd $($(U_HL_DEV)_DEPLOY_DIR);					\
+	rm -rf $($(U_HL_DEV)_WORK_DIR)/workspace/sources/$(patsubst	\
+	openbmc-%-remove,%,$@);						\
+	rm -rf $($(U_HL_DEV)_WORK_DIR)/workspace/recipes/$(patsubst	\
+	openbmc-%-remove,%,$@);						\
+	rm -f $($(U_HL_DEV)_WORK_DIR)/workspace/appends/$(patsubst	\
+	openbmc-%-remove,%,$@)*
diff --git a/scripts/make/hls/Makefile b/scripts/make/hls/Makefile
index f38ff71..0bfbb01 100644
--- a/scripts/make/hls/Makefile
+++ b/scripts/make/hls/Makefile
@@ -1,22 +1,26 @@
 export OPENBMC_WORK_DIR = build
-export OPENBMC_ENV_FLAGS = meta-habana/meta-hls/conf
+export HLS_ENV_FLAGS = meta-habana/meta-hls/conf
 export OPENBMC_IMAGE_NAME = flash-hls
 export OBMC_PHOSPHOR_IMAGE_BASENAME = obmc-phosphor-image-hls
 export OBMC_PHOSPHOR_IMAGE_NAME = $(OBMC_PHOSPHOR_IMAGE_BASENAME).static.mtd.tar
 export OBMC_PHOSPHOR_FULLCHIP_IMAGE_NAME = $(OBMC_PHOSPHOR_IMAGE_BASENAME).static.mtd.all.tar
 export OPENBMC_IMAGE_DIR = $(OPENBMC_WORK_DIR)/tmp/deploy/images/hls
-export OPENBMC_DEPLOY_DIR = $(OUTPUT_DIR)/build/openbmc-custom
+export HLS_DEPLOY_DIR = $(OUTPUT_DIR)/build/openbmc-custom
 unexport TEMPLATECONF
 
 ABS_OPENBMC_WORK_DIR = $(OPENBMC_DEPLOY_DIR)/$(OPENBMC_WORK_DIR)
 OPENBMC_DEVTOOL_SOURCES = $(ABS_OPENBMC_WORK_DIR)/workspace/sources
 
-define OPENBMC_BITBAKE_ENV
+ifeq ($(HL_DEV),hls)
+	include scripts/make/common/Makefile.openbmc
+endif
+
+define HLS_BITBAKE_ENV
 	BB_ENV_EXTRAWHITE+=" HL_VERSION HABANA_SOFTWARE_STACK HL_LINUX_ROOT HOME BUILD_ROOT OPENBMC_BUILD_CACHE"
 endef
 export OPENBMC_BITBAKE_ENV
 
-define OPENBMC_ENV
+define HLS_ENV
 	mkdir -p $(OPENBMC_WORK_DIR);				\
 	TEMPLATECONF=$(OPENBMC_ENV_FLAGS) source ./openbmc-env;
 endef
@@ -73,133 +77,11 @@
 	@echo '***************HLS ASIC INSTALL***************'
 	sshpass -p$(HL_PASSWD) $(SCP) $(IMAGES_DIR)/flash-hls-pkg.tgz	root@$(HL_INSTALL_LAB_IP):/tmp/
 
-
 .PHONY: hls-asic-robot
 hls-asic-robot: hls-asic-prepare
 	@echo '***************HLS ASIC ROBOT***************'
 	$(MAKE) common-asic-robot
 
-
-
-
-hls-help:
-	@echo
-	@echo 'OpenBMC bitbake & devtool wrappers for Habana Labs server build'
-	@echo '---------------------------------------------------------------'
-	@echo
-	@echo 'OpenBMC devtool commands can be used for adding new'
-	@echo ' packages and modifying existing ones, these targets wrap its'
-	@echo ' functionality:'
-	@echo
-	@echo '1. hls-help:'
-	@echo '   Print this help menu.'
-	@echo '2. openbmc-add:'
-	@echo '   Adds a new package from source, and generates a recipe for it.'
-	@echo '   Usage*:'
-	@echo '   $ make openbmc-add URI="<git://[...]>"'
-	@echo '   * In case of ssh repos, ";protocol=ssh" should be added as a'
-	@echo '     suffix.'
-	@echo '3. openbmc-<pkg-name>-modify:'
-	@echo '   Deployes a package already managed by OpenBMC and opens a'
-	@echo '   terminal in its sources path.'
-	@echo '   From now on, each change in the sources will trigger'
-	@echo '   a new build when calling 'make hls-build'.'
-	@echo '   Usage:'
-	@echo '   $ make openbmc-<pkg-name>-modify'
-	@echo '4. openbmc-<pkg-name>-menuconfig:'
-	@echo '   Opens a 'menuconfig' session for the package'
-	@echo '   (if it has a menuconfig target).'
-	@echo '   Usage:'
-	@echo '   $ make openbmc-<pkg-name>-menuconfig'
-	@echo '5. openbmc-<pkg-name>-build:'
-	@echo '   Builds a package that is already deployed via'
-	@echo '   openbmc-<pkg-name>-modify.'
-	@echo '   Usage:'
-	@echo '   $ make openbmc-<pkg-name>-build'
-	@echo '6. openbmc-<pkg-name>-clean:'
-	@echo '   Cleans the build artifacts of the package.'
-	@echo '   Usage:'
-	@echo '   $ make openbmc-<pkg-name>-clean'
-	@echo '7. openbmc-<pkg-name>-remove*:'
-	@echo '   Clears the package files from the local environment*.'
-	@echo '   Usage:'
-	@echo '   $ make openbmc-<pkg-name>-remove'
-	@echo '   * The sources dir will be deleted, so one needs to make'
-	@echo '     sure his work was pushed before running it.'
-	@echo '8. openbmc-<pkg-name>-diffconfig:'
-	@echo '   Opens a 'diffconfig' session for the package'
-	@echo '   (if it has a diffconfig target).'
-	@echo '   Usage:'
-	@echo '   $ make openbmc-<pkg-name>-diffconfig'
-
 .PHONY: hls-distclean
 hls-distclean: hl-hls
 	@$(MAKE) hl-distclean
-
-inner-openbmc-add:
-	@cd $(OPENBMC_DEPLOY_DIR);					\
-	$(OPENBMC_ENV) $(OPENBMC_BITBAKE_ENV)				\
-	devtool add "$(URI)"
-
-openbmc-add: hls-asic-prepare
-	@$(MAKE) inner-$@
-
-inner-openbmc-%-modify:
-	@cd $(OPENBMC_DEPLOY_DIR);					\
-	$(OPENBMC_ENV) $(OPENBMC_BITBAKE_ENV)				\
-	devtool modify $(patsubst inner-openbmc-%-modify,%,$@);	\
-	terminator							\
-	--working-directory="$(OPENBMC_DEVTOOL_SOURCES)/$(patsubst	\
-		inner-openbmc-%-modify,%,$@)" &
-
-openbmc-%-modify: hls-asic-prepare
-	@$(MAKE) inner-$@
-
-inner-openbmc-%-menuconfig:
-	@cd $(OPENBMC_DEPLOY_DIR);					\
-	$(OPENBMC_ENV) $(OPENBMC_BITBAKE_ENV)				\
-	bitbake -c menuconfig $(patsubst inner-openbmc-%-menuconfig,%,$@)
-
-openbmc-%-menuconfig: hls-asic-prepare
-	@$(MAKE) inner-$@
-
-inner-openbmc-%-diffconfig:
-	@cd $(OPENBMC_DEPLOY_DIR);					\
-	$(OPENBMC_ENV) $(OPENBMC_BITBAKE_ENV)				\
-	bitbake -c diffconfig $(patsubst inner-openbmc-%-diffconfig,%,$@)
-
-openbmc-%-diffconfig: hls-asic-prepare
-	@$(MAKE) inner-$@
-
-inner-openbmc-%-build:
-	@cd $(OPENBMC_DEPLOY_DIR);					\
-	$(OPENBMC_ENV) $(OPENBMC_BITBAKE_ENV)				\
-	devtool build $(patsubst inner-openbmc-%-build,%,$@)
-
-openbmc-%-build: hls-asic-prepare
-	@$(MAKE) inner-$@
-
-inner-openbmc-%-clean:
-	@cd $(OPENBMC_DEPLOY_DIR);					\
-	$(OPENBMC_ENV) $(OPENBMC_BITBAKE_ENV)				\
-	bitbake -c clean $(patsubst inner-openbmc-%-clean,%,$@)
-
-openbmc-%-clean: hls-asic-prepare
-	@$(MAKE) inner-$@
-
-inner-openbmc-%-cleanall:
-	@cd $(OPENBMC_DEPLOY_DIR);					\
-	$(OPENBMC_ENV) $(OPENBMC_BITBAKE_ENV)				\
-	bitbake -c cleanall $(patsubst inner-openbmc-%-cleanall,%,$@)
-
-openbmc-%-cleanall: hls-asic-prepare
-	@$(MAKE) inner-$@
-
-openbmc-%-remove:
-	@cd $(OPENBMC_DEPLOY_DIR);					\
-	rm -rf $(OPENBMC_WORK_DIR)/workspace/sources/$(patsubst		\
-	openbmc-%-remove,%,$@);						\
-	rm -rf $(OPENBMC_WORK_DIR)/workspace/recipes/$(patsubst		\
-	openbmc-%-remove,%,$@);						\
-	rm -f $(OPENBMC_WORK_DIR)/workspace/appends/$(patsubst		\
-	openbmc-%-remove,%,$@)*
